module_test_ios failing on M1, 'FlutterUITests' not valid for use in process: Trying to load an unsigned library

```
[module_test_ios] [STDOUT] stderr: 	FlutterUITests-Runner (8236) encountered an error (Failed to load the test bundle. If you believe this error represents a bug, please attach the result bundle at /opt/s/w/ir/x/t/flutter_module_test_ios_xcresult. (Underlying Error: The bundle “FlutterUITests” couldn’t be loaded. The bundle couldn’t be loaded. Try reinstalling the bundle. 
...
'/Users/chrome-bot/Library/Developer/CoreSimulator/Devices/AE118AD8-EA4E-45A7-91E7-029FA264FB15/data/Containers/Bundle/Application/B2A19E15-4DDF-4DD1-8B1D-742D5607AFB5/FlutterUITests-Runner.app/PlugIns/FlutterUITests.xctest/FlutterUITests' not valid for use in process: Trying to load an unsigned library)))
```
https://ci.chromium.org/p/flutter/builders/try/Mac%20module_test_ios/18504

Discovered in https://github.com/flutter/flutter/pull/109889

> @dcharkes Can you try commenting out these lines and see if that fixes it for you locally?
> 
> https://github.com/flutter/flutter/blob/bae1ac2f6f35a1d49fd826287d5f845e9717eaa2/dev/devicelab/bin/tasks/module_test_ios.dart#L492-L495

That leads to: `Provisioning profile "iOS Team Provisioning Profile: *" doesn't include the currently selected device`

```
[2023-07-26 20:27:36.322387] [STDOUT] stdout: /var/folders/2y/mngq9h194yzglt4kzttzfq6800klzg/T/flutter_module_test.1W4lHO/hello_host_app/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.1.99. (in target 'GTMSessionFetcher' from project 'Pods')
[2023-07-26 20:27:36.322632] [STDOUT] stdout: /var/folders/2y/mngq9h194yzglt4kzttzfq6800klzg/T/flutter_module_test.1W4lHO/hello_host_app/Host.xcodeproj: error: Provisioning profile "iOS Team Provisioning Profile: *" doesn't include the currently selected device "dacoharkes-macbookpro2" (identifier 00006000-001C30492ED2801E). (in target 'Host' from project 'Host')
[2023-07-26 20:27:36.322657] [STDOUT] stdout: warning: Run script build phase '[CP-User] Run Flutter Build hello Script' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Host' from project 'Host')
[2023-07-26 20:27:36.322664] [STDOUT] stdout: warning: Run script build phase '[CP-User] Embed Flutter Build hello Script' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Host' from project 'Host')
[2023-07-26 20:27:36.322670] [STDOUT] stdout: /var/folders/2y/mngq9h194yzglt4kzttzfq6800klzg/T/flutter_module_test.1W4lHO/hello_host_app/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.1.99. (in target 'AppAuth' from project 'Pods')
[2023-07-26 20:27:36.322676] [STDOUT] stdout: /var/folders/2y/mngq9h194yzglt4kzttzfq6800klzg/T/flutter_module_test.1W4lHO/hello_host_app/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.1.99. (in target 'GoogleSignIn-GoogleSignIn' from project 'Pods')
[2023-07-26 20:27:36.322696] [STDOUT] stdout: /var/folders/2y/mngq9h194yzglt4kzttzfq6800klzg/T/flutter_module_test.1W4lHO/hello_host_app/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.1.99. (in target 'GoogleSignIn' from project 'Pods')
[2023-07-26 20:27:36.322793] [STDOUT] stdout: /var/folders/2y/mngq9h194yzglt4kzttzfq6800klzg/T/flutter_module_test.1W4lHO/hello_host_app/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.1.99. (in target 'GTMAppAuth' from project 'Pods')
[2023-07-26 20:27:36.324182] [STDOUT] stderr: ** BUILD FAILED **
```

--- a/.ci.yaml
+++ b/.ci.yaml
@@ -3884,7 +3884,6 @@ targets:
 
   - name: Mac_arm64 module_test_ios
     recipe: devicelab/devicelab_drone
-    bringup: true
     timeout: 60
     properties:
       dependencies: >-
@@ -3893,20 +3892,6 @@ targets:
         ]
       tags: >
         ["devicelab", "hostonly", "mac"]
-      task_name: module_test_ios
-    runIf:
-      - dev/**
-      - packages/flutter_tools/**
-      - bin/**
-      - .ci.yaml
-
-  - name: Mac_arm64_ios module_test_ios # Must be run on devicelab bot for codesigning https://github.com/flutter/flutter/issues/112033
-    recipe: devicelab/devicelab_drone
-    presubmit: false
-    timeout: 60
-    properties:
-      tags: >
-        ["devicelab", "ios", "mac", "arm64"]
       task_name: module_test_ios
     runIf:
       - dev/**

--- a/.ci.yaml
+++ b/.ci.yaml
@@ -3864,24 +3864,6 @@ targets:
       - bin/**
       - .ci.yaml
 
-  - name: Mac_x64 module_test_ios
-    bringup: true # Flaky https://github.com/flutter/flutter/issues/144680
-    recipe: devicelab/devicelab_drone
-    timeout: 60
-    properties:
-      dependencies: >-
-        [
-          {"dependency": "ruby", "version": "ruby_3.1-pod_1.13"}
-        ]
-      tags: >
-        ["devicelab", "hostonly", "mac"]
-      task_name: module_test_ios
-    runIf:
-      - dev/**
-      - packages/flutter_tools/**
-      - bin/**
-      - .ci.yaml
-
   - name: Mac_arm64 module_test_ios
     recipe: devicelab/devicelab_drone
     timeout: 60

